### PR TITLE
Deployment Configs - Follow-ups

### PR DESCRIPTION
**Pull request by makungaj1**
*Issue #, if available:*

*Description of changes:*
* Init Deployment config outside Model init
* ModelBuilder set deployment to accept both deployment config name and instance type
* Display deployment config benchmark metrics to display multiple raw.

*Testing done:*
Unit tests and Notebooks E2E

```
jumpstart_model = JumpStartModel(model_id="meta-textgeneration-llama-2-7b-f")

jumpstart_model.list_deployment_configs()
```

Output

```
[{'DeploymentConfigName': 'lmi',
  'DeploymentArgs': {'ImageUri': '763104351884.dkr.ecr.us-west-2.amazonaws.com/djl-inference:0.27.0-deepspeed0.12.6-cu121',
   'ModelData': {'S3DataSource': {'S3Uri': 's3://jumpstart-private-cache-prod-us-west-2/meta-textgeneration/meta-textgeneration-llama-2-7b-f/artifacts/inference-prepack/v2.0.0/',
     'S3DataType': 'S3Prefix',
     'CompressionType': 'None'}},
   'Environment': {'SAGEMAKER_PROGRAM': 'inference.py',
    'ENDPOINT_SERVER_TIMEOUT': '3600',
    'MODEL_CACHE_ROOT': '/opt/ml/model',
    'SAGEMAKER_ENV': '1',
    'HF_MODEL_ID': '/opt/ml/model',
    'MAX_INPUT_LENGTH': '4095',
    'MAX_TOTAL_TOKENS': '4096',
    'SAGEMAKER_MODEL_SERVER_WORKERS': '1'},
   'InstanceType': 'ml.g5.12xlarge',
   'ComputeResourceRequirements': {'MinMemoryRequiredInMb': 98304,
    'NumberOfAcceleratorDevicesRequired': 4},
   'ModelDataDownloadTimeout': 1200,
   'ContainerStartupHealthCheckTimeout': 1200},
  'AccelerationConfigs': None,
  'BenchmarkMetrics': {'ml.g5.2xlarge': [{'name': 'min_latency',
     'value': '34',
     'unit': 'ms'},
    {'name': 'max_throughput', 'value': '173', 'unit': 'ms'},
    {'name': 'max_concurrency', 'value': '64', 'unit': 'count'},
    {'name': 'Instance Rate', 'value': '1.515', 'unit': 'USD/Hrs'}],
   'ml.g5.12xlarge': [{'name': 'min_latency', 'value': '12', 'unit': 'ms'},
    {'name': 'max_throughput', 'value': '294', 'unit': 'ms'},
    {'name': 'max_concurrency', 'value': '64', 'unit': 'count'},
    {'name': 'Instance Rate', 'value': '7.09', 'unit': 'USD/Hrs'}],
   'ml.p4d.24xlarge': [{'name': 'min_latency', 'value': '7', 'unit': 'ms'},
    {'name': 'max_throughput', 'value': '1414', 'unit': 'ms'},
    {'name': 'max_concurrency', 'value': '256', 'unit': 'count'},
    {'name': 'Instance Rate', 'value': '38.67', 'unit': 'USD/Hrs'}]}},
 {'DeploymentConfigName': 'lmi-trtllm',
  'DeploymentArgs': {'ImageUri': '763104351884.dkr.ecr.us-west-2.amazonaws.com/djl-inference:0.27.0-deepspeed0.12.6-cu121',
   'ModelData': {'S3DataSource': {'S3Uri': 's3://jumpstart-private-cache-prod-us-west-2/meta-textgeneration/meta-textgeneration-llama-2-7b-f/artifacts/inference-prepack/v2.0.0/',
     'S3DataType': 'S3Prefix',
     'CompressionType': 'None'}},
   'Environment': {'SAGEMAKER_PROGRAM': 'inference.py',
    'ENDPOINT_SERVER_TIMEOUT': '3600',
    'MODEL_CACHE_ROOT': '/opt/ml/model',
    'SAGEMAKER_ENV': '1',
    'HF_MODEL_ID': '/opt/ml/model',
    'MAX_INPUT_LENGTH': '4095',
    'MAX_TOTAL_TOKENS': '4096',
    'SAGEMAKER_MODEL_SERVER_WORKERS': '1'},
   'InstanceType': 'ml.g5.12xlarge',
   'ComputeResourceRequirements': {'MinMemoryRequiredInMb': 98304,
    'NumberOfAcceleratorDevicesRequired': 4},
   'ModelDataDownloadTimeout': 1200,
   'ContainerStartupHealthCheckTimeout': 1200},
  'AccelerationConfigs': None,
  'BenchmarkMetrics': {'ml.g5.2xlarge': [{'name': 'min_latency',
     'value': '34',
     'unit': 'ms'},
    {'name': 'max_throughput', 'value': '173', 'unit': 'ms'},
    {'name': 'max_concurrency', 'value': '64', 'unit': 'count'},
    {'name': 'Instance Rate', 'value': '1.515', 'unit': 'USD/Hrs'}],
   'ml.g5.12xlarge': [{'name': 'min_latency', 'value': '12', 'unit': 'ms'},
    {'name': 'max_throughput', 'value': '294', 'unit': 'ms'},
    {'name': 'max_concurrency', 'value': '64', 'unit': 'count'},
    {'name': 'Instance Rate', 'value': '7.09', 'unit': 'USD/Hrs'}],
   'ml.p4d.24xlarge': [{'name': 'min_latency', 'value': '7', 'unit': 'ms'},
    {'name': 'max_throughput', 'value': '1414', 'unit': 'ms'},
    {'name': 'max_concurrency', 'value': '256', 'unit': 'count'},
    {'name': 'Instance Rate', 'value': '38.67', 'unit': 'USD/Hrs'}]}},
 {'DeploymentConfigName': 'tgi',
  'DeploymentArgs': {'ImageUri': '763104351884.dkr.ecr.us-west-2.amazonaws.com/djl-inference:0.27.0-deepspeed0.12.6-cu121',
   'ModelData': {'S3DataSource': {'S3Uri': 's3://jumpstart-private-cache-prod-us-west-2/meta-textgeneration/meta-textgeneration-llama-2-7b-f/artifacts/inference-prepack/v2.0.0/',
     'S3DataType': 'S3Prefix',
     'CompressionType': 'None'}},
   'Environment': {'SAGEMAKER_PROGRAM': 'inference.py',
    'ENDPOINT_SERVER_TIMEOUT': '3600',
    'MODEL_CACHE_ROOT': '/opt/ml/model',
    'SAGEMAKER_ENV': '1',
    'HF_MODEL_ID': '/opt/ml/model',
    'MAX_INPUT_LENGTH': '4095',
    'MAX_TOTAL_TOKENS': '4096',
    'SAGEMAKER_MODEL_SERVER_WORKERS': '1'},
   'InstanceType': 'ml.g5.12xlarge',
   'ComputeResourceRequirements': {'MinMemoryRequiredInMb': 98304,
    'NumberOfAcceleratorDevicesRequired': 4},
   'ModelDataDownloadTimeout': 1200,
   'ContainerStartupHealthCheckTimeout': 1200},
  'AccelerationConfigs': None,
  'BenchmarkMetrics': {'ml.g5.2xlarge': [{'name': 'min_latency',
     'value': '33',
     'unit': 'ms'},
    {'name': 'max_throughput', 'value': '121', 'unit': 'ms'},
    {'name': 'max_concurrency', 'value': '8', 'unit': 'count'},
    {'name': 'Instance Rate', 'value': '1.515', 'unit': 'USD/Hrs'}],
   'ml.g5.12xlarge': [{'name': 'min_latency', 'value': '13', 'unit': 'ms'},
    {'name': 'max_throughput', 'value': '306', 'unit': 'ms'},
    {'name': 'max_concurrency', 'value': '64', 'unit': 'count'},
    {'name': 'Instance Rate', 'value': '7.09', 'unit': 'USD/Hrs'}],
   'ml.p4d.24xlarge': [{'name': 'min_latency', 'value': '8', 'unit': 'ms'},
    {'name': 'max_throughput', 'value': '1714', 'unit': 'ms'},
    {'name': 'max_concurrency', 'value': '512', 'unit': 'count'},
    {'name': 'Instance Rate', 'value': '38.67', 'unit': 'USD/Hrs'}]}}]
```

```
jumpstart_model.display_benchmark_metrics()
```

Output

```
| Config Name   | Instance Type            |   Instance Rate (USD/Hrs) |   Min Latency (ms) |   Max Throughput (ms) |   Max Concurrency (count) |
|:--------------|:-------------------------|--------------------------:|-------------------:|----------------------:|--------------------------:|
| lmi           | ml.g5.12xlarge (Default) |                     7.09  |                 12 |                   294 |                        64 |
| lmi           | ml.g5.2xlarge            |                     1.515 |                 34 |                   173 |                        64 |
| lmi           | ml.p4d.24xlarge          |                    38.67  |                  7 |                  1414 |                       256 |
| lmi-trtllm    | ml.g5.12xlarge (Default) |                     7.09  |                 12 |                   294 |                        64 |
| lmi-trtllm    | ml.g5.2xlarge            |                     1.515 |                 34 |                   173 |                        64 |
| lmi-trtllm    | ml.p4d.24xlarge          |                    38.67  |                  7 |                  1414 |                       256 |
| tgi           | ml.g5.12xlarge (Default) |                     7.09  |                 13 |                   306 |                        64 |
| tgi           | ml.g5.2xlarge            |                     1.515 |                 33 |                   121 |                         8 |
| tgi           | ml.p4d.24xlarge          |                    38.67  |                  8 |                  1714 |                       512 |

```

```
model_builder = ModelBuilder(
    model="meta-textgeneration-llama-2-7b-f",
    schema_builder=SchemaBuilder(sample_input, sample_output),
)

model_builder.list_deployment_configs()

model_builder.display_benchmark_metrics()

model_builder.set_deployment_config("lmi", "ml.g5.12xlarge")
```

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/jumpstart/model.py [hunk 1]
```diff
@@ -449,33 +444,46 @@ def set_deployment_config(self, config_name: str, instance_type: str) -> None:
...
         Returns:
-            Metrics: Pandas DataFrame object.
+            Benchmark Metrics: Pandas DataFrame object.
         """
-        return pd.DataFrame(self._get_benchmarks_data(self.config_name))
+        benchmark_metrics_data = self._get_deployment_configs_benchmarks_data(
+            self.config_name, self.instance_type
+        )
+        keys = list(benchmark_metrics_data.keys())
+        df = pd.DataFrame(benchmark_metrics_data).sort_values(by=[keys[0], keys[1]])
```
Comment (qiyunz) [inline src/sagemaker/jumpstart/model.py:468]: nit: add comments on the sorting columns

**Review by qiyunz — APPROVED** [review]

File: src/sagemaker/jumpstart/model.py [hunk 2]
```diff
@@ -866,92 +874,94 @@ def register_deploy_wrapper(*args, **kwargs):
...
-        if should_fetch_instance_rate_metric:
-            instance_rate = get_instance_rate_per_hour(
-                instance_type=default_inference_instance_type, region=self.region
+        deployment_configs = []
+        if self._metadata_configs is None:
+            return deployment_configs
+
+        err = None
+        for config_name, metadata_config in self._metadata_configs.items():
+            if err is None or "is not authorized to perform: pricing:GetProducts" not in err:
```
Comment (jiapinw) [inline src/sagemaker/jumpstart/model.py:928]: Is this only permission required?

Comment (jiapinw) [reply src/sagemaker/jumpstart/model.py:928]: Not quite following the logic here. Say in the first iteration err is assigned with no pricing:GetProducts permission exception, is there a need to continue the rest of this function?

**Review by jiapinw — APPROVED** [review]

File: src/sagemaker/jumpstart/model.py (hunk 2 above)
Comment (jiapinw) [reply src/sagemaker/jumpstart/model.py:928]: Discussed offline, cleaner logic will be applied in following prs.

Comment (makungaj1) [reply src/sagemaker/jumpstart/model.py:928]: We continue the rest but the only thing we skip is `add_instance_rate_stats_to_benchmark_metrics` if err contains `is not authorized to perform: pricing:GetProducts`

I'll clean it up to make it clear.

**Review by zhaoqizqwang — APPROVED** [review]

**Pull request merged**